Epsilon.Host.Frontend/src/components/competence/CompetenceDocument.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+	<div v-if="submissions" class="competence-document">
+		<KpiTable :submissions="submissions" :outcomes="allOutcomes" />
+		<div></div>
+		<div></div>
+		<CompetenceProfile
+			v-if="hboIDomain"
+			:submissions="submissions"
+			:domain="hboIDomain" />
+		<div></div>
+		<div></div>
+		<KpiMatrix
+			v-if="hboIDomain"
+			:submissions="submissions"
+			:domain="hboIDomain" />
+	</div>
+</template>
+
+<script lang="ts" setup>
+import KpiMatrix from "~/components/competence/KpiMatrix.vue"
+import KpiTable from "~/components/competence/KpiTable.vue"
+import CompetenceProfile from "~/components/competence/CompetenceProfile.vue"
+import type {
+	LearningDomain,
+	LearningDomainSubmission,
+	LearningDomainOutcome,
+} from "~/api.generated"
+
+const props = defineProps<{
+	submissions: LearningDomainSubmission[]
+}>()
+
+const api = useApi()
+
+const hboIDomain = ref<LearningDomain | null>(null)
+const personalDevelopmentDomain = ref<LearningDomain | null>(null)
+
+onMounted(async () => {
+	hboIDomain.value = (await api.learning.domainDetail("hbo-i-2018")).data
```

Comment:
Please don't use await in this case. Please use `then`. Now the functions will be triggered after each other, but they can be trigger at the same time. It makes the application load faster.